closes issue #4: Restore not working in AWS and Openstack

pritishmishra opened the pull request:
Description: Modified the logic of copying data from encrypted backups to fix the bug in restore functionality.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/service-fabrik-blueprint-service?pullRequest=5) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/service-fabrik-blueprint-service?pullRequest=5) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/service-fabrik-blueprint-service?pullRequest=5) it.</sub>

Commit:
Closes-Bug: issues: #4: Restore not working in AWS and Openstack
Description: Modified the logic of copying data from encrypted backups to fix the bug in restore functionality.

kamath-prasad commented on backuprestore/restore.py:
```diff
@@ -159,12 +159,8 @@ def main():
-                iaas_client.exit('Could not copy from {}/{} to the persistent volume.'
-                                 .format(DIRECTORY_DOWNLOADS, DIRECTORY_PERSISTENT))
-
+	          if not iaas_client.shell('cp -r {}/blueprint/files/* {}/blueprint/files; exit 0'.format(DIRECTORY_DOWNLOADS, DIRECTORY_PERSISTENT)):
```
as discussed please check if the folder is empty first - 

if yes , exit 0
if no , 'cp -r {}/blueprint/files/* {}/blueprint/files;'

sauravmndl-zz replied:
+1

sauravmndl-zz replied:
use to check empty dir or not

`ls -A <path>/blueprint/files/`

which ignore . and ..

ls --help
-A, --almost-all           do not list implied . and ..

sauravmndl-zz replied:
Do as follows: 

if not iaas_client.shell('ls -A /var/vcap/store/blueprint/files/').rstrip():
         skip copy
else:
         copy


# rstrip() is to remove trailing \n \r space etc....

pritishmishra replied:
ok..
I tried with 'ls -A' and rsync and was somehow unable to make it work.
So, I used os.listdir() which returns empty array ([]) if the list is empty, i.e., for empty folders it will return [] and hence, for non-empty folders only, the copy operation will occur.
Hope this is fine.

sauravmndl-zz replied:
yes. if works ... go ahead

kamath-prasad replied:
+1 .

@pritishmishra  : please talk to @akskap and create a blueprint boshrelease and use the new version in sf product component.

sauravmndl-zz reviewed (changes requested):
Make the above changes

Commit:
Closes-Bug: issues: #4: Restore not working in AWS and Openstack

Description: Implemented Review comments
